DaS_Programs/Singly linked list/Singly linked list/main.c

```diff
@@ -356,14 +362,32 @@ struct node *delete_before(struct node *start)
         
         preptr=start;
         ptr=start->next;
-        
-        while (ptr->next->data!=val)
+        if(preptr->data==val)
```

Comment:
if the start nodes data is indeed equal then, the node still needs to be deleted.
Changes: 

```
if(preptr->data==val)
{
     //call the *delete_list 
}
```

Reply:
you sure it is necesssary to delete in this case , since there is no node before the first one?? if yes then we can delete only the first node then instead of the entire list

Reply:
So since in this case the first node is the one that has the same data as value, then we need to delete this node. So instead of writing the delete function again, we just call the *delete_list and it'll be lesser code to write.